src/Compilers/Core/Portable/FileSystem/FileUtilities.cs

```diff
@@ -372,6 +372,21 @@ internal static DateTime GetFileTimeStamp(string fullPath)
             }
         }
 
+        /// <exception cref="IOException"/>
+        internal static long GetFileLength(string fullPath)
+        {
+            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
+            try
+            {
+                var info = new FileInfo(fullPath);
+                return info.Length;
+            }
+            catch (Exception e)
+            {
+                throw new IOException(e.Message, e);
```

Comment:
Sure.  We are diligent to the places we know about and that's the problem.  There are plenty of more cases where IO occurs that we don't know about.  Every miss is just a crash that we're passing onto the customer instead of fixing now.  

This is why I'm always very skeptical of approaches like this.  Every time a customer reports the exception we will just catch ,wrap in IOException and move on.  Hence the end behavior is indistinguishable from just catching Exception in the first place.  This approach just causes us more work.